service_taxi/src/main/kotlin/org/thechance/service_taxi/api/endpoints/TaxiRoutes.kt

```diff
@@ -2,31 +2,66 @@ package org.thechance.service_taxi.api.endpoints
 
 import io.ktor.http.HttpStatusCode
 import io.ktor.server.application.call
+import io.ktor.server.request.receive
 import io.ktor.server.response.respond
 import io.ktor.server.routing.Route
+import io.ktor.server.routing.delete
 import io.ktor.server.routing.get
 import io.ktor.server.routing.post
+import io.ktor.server.routing.put
 import io.ktor.server.routing.route
 import org.koin.ktor.ext.inject
+import org.thechance.service_taxi.api.models.TaxiDto
 import org.thechance.service_taxi.api.models.toDto
-import org.thechance.service_taxi.domain.entity.Taxi
-import org.thechance.service_taxi.domain.usecase.AddTaxiUseCase
-import org.thechance.service_taxi.domain.usecase.GetAllTaxesUseCase
+import org.thechance.service_taxi.api.models.toTaxi
+import org.thechance.service_taxi.domain.usecase.TaxiUseCasesContainer
 
 fun Route.taxiRoutes() {
-    val getAllTaxesUseCase: GetAllTaxesUseCase by inject()
-    val addTaxiUseCase: AddTaxiUseCase by inject()
+    val taxiUseCasesContainer: TaxiUseCasesContainer by inject()
 
     route("/taxi") {
         get {
-            val result = getAllTaxesUseCase()
+            val result = taxiUseCasesContainer.getAllTaxesUseCase()
             call.respond(HttpStatusCode.OK, result.toDto())
         }
 
+        get("/{taxiId}") {
+            val id = call.parameters["taxiId"]?.trim().orEmpty()
+            val result = taxiUseCasesContainer.getTaxiByIdUseCase(id)
+
+            result?.let { call.respond(HttpStatusCode.OK, result.toDto()) }
+                ?: call.respond(HttpStatusCode.NotFound, "taxi not found")
+        }
+
         post {
-            val name = call.parameters["name"]?.trim()
-            val result = addTaxiUseCase(Taxi(name = name))
-            call.respond(HttpStatusCode.OK, result)
+            val taxi = call.receive<TaxiDto>()
+            val result = taxiUseCasesContainer.addTaxiUseCase(taxi.toTaxi())
+            if (result) {
```

Comment:
`if (result) {...`
this is should be handle it in usecase i think

Reply:
I didn't got it